src/scmrepo/git/credentials.py

```diff
@@ -283,6 +283,8 @@ def _input_tty(prompt: str = "Username: ") -> str:
     from contextlib import ExitStack
 
     if os.name == "nt":
+        if not sys.stdout.isatty():
```

Comment:
We should be looking for `sys.stdin.isatty()` not `sys.stdout`, no?

Reply:
Hmm, I guess technically that's probably more correct, but I'm not sure we should be trying to read from stdin if we can't actually write out the prompt either?